### PR TITLE
Set completionList only as incomplete if the request has been cancell…

### PR DESCRIPTION
**Pull request by rubenporras**
…ed or if the proposal acceptor could not accept all the proposals because the limit of proposals to accept has been reached.

### CODE REVIEW

**Comment by rubenporras** [comment]
for https://github.com/eclipse/xtext-core/issues/1999

File: org.eclipse.xtext.ide/src/org/eclipse/xtext/ide/editor/contentassist/IdeContentProposalProvider.java [hunk 1]
```diff
@@ -61,16 +61,18 @@ public class IdeContentProposalProvider {
 
 	/**
 	 * Create content assist proposals and pass them to the given acceptor.
+	 * @return true if the acceptor accepted all the proposals, false otherwise.
 	 */
-	public void createProposals(Collection<ContentAssistContext> contexts, IIdeContentProposalAcceptor acceptor) {
```
Comment (szarnekow) [inline org.eclipse.xtext.ide/src/org/eclipse/xtext/ide/editor/contentassist/IdeContentProposalProvider.java:65]: This is not binary compatible. Do you strictly need the boolean return value or would it be possible to communicate this differently?

Comment (rubenporras) [reply org.eclipse.xtext.ide/src/org/eclipse/xtext/ide/editor/contentassist/IdeContentProposalProvider.java:65]: I did not think about binary compatibility. 

One alternative I can think about would be to add a method `didAcceptedAllProposals` to `IIdeContentProposalAcceptor`. That would not break binary compatibility as far as I understand. If we also add a default implementation, it should not break source compatibility either, the problem three is that I would not know if we should return false or true.

So to keep both probably the cleanest would be to add a new interface `IIdeContentProposalAcceptor2` that inherit from `IIdeContentProposalAcceptor` and adds a `didAcceptedAllProposals`. Should I go for that one?

**Comment by rubenporras** [comment]
@szarnekow , any more comments on this one? I think it would be ready to merge,

File: org.eclipse.xtext.ide/src/org/eclipse/xtext/ide/editor/contentassist/IIdeContentProposalAcceptor.java [hunk 2]
```diff
@@ -20,4 +20,15 @@ public interface IIdeContentProposalAcceptor {
 	
 	boolean canAcceptMoreProposals();
 	
+	/**
+	 * If the acceptor accepted all proposals.
+	 * 
+	 * @since 2.29
+	 */
+	default boolean couldAcceptAllProposals() {
+		// this default implentation will be accurate for all cases but the one
```
Comment (szarnekow) [inline org.eclipse.xtext.ide/src/org/eclipse/xtext/ide/editor/contentassist/IIdeContentProposalAcceptor.java:29]: Typo in `implentation`

File: org.eclipse.xtext.ide/src/org/eclipse/xtext/ide/editor/contentassist/IIdeContentProposalAcceptor.java [hunk 3]
```diff
@@ -20,4 +20,15 @@ public interface IIdeContentProposalAcceptor {
 	
 	boolean canAcceptMoreProposals();
 	
+	/**
+	 * If the acceptor accepted all proposals.
+	 * 
+	 * @since 2.29
+	 */
+	default boolean couldAcceptAllProposals() {
```
Comment (szarnekow) [inline org.eclipse.xtext.ide/src/org/eclipse/xtext/ide/editor/contentassist/IIdeContentProposalAcceptor.java:28]: Maybe call it `didAcceptAllProposals`

**Review by szarnekow — APPROVED** [review]
Two small things that I'd like to ask you to fix.

**Review by szarnekow — APPROVED** [review]

**Pull request merged**